samcli/lib/test_runner/test_runner_template_generator.py

```diff
@@ -0,0 +1,255 @@
+        "  Action:\n"
+        "   {%- for action in action_list %}\n"
+        "     # - {{action}}\n"
+        "   {%- endfor %}\n"
+        "  Resource: {{arn}}\n"
+    )
+    mapping = _get_action_mapping(resource_arn)
+
+    if mapping is None:
+        return jinja2.Template(new_statement_template).render(action_list=[], arn=resource_arn).rstrip()
+
+    if mapping["Resource"] in ("AWS::ApiGateway::Api", "AWS::ApiGateway::RestApi"):
+        apiId = _extract_api_id_from_arn(resource_arn)
+        execute_api_arn = f"!Sub arn:${{AWS::Partition}}:execute-api:${{AWS::Region}}:${{AWS::AccountId}}:{apiId}/<STAGE>/GET/<RESOURCE_PATH>"
+        return (
+            jinja2.Template(new_statement_template).render(action_list=mapping["Action"], arn=execute_api_arn).rstrip()
+        )
+
+    return jinja2.Template(new_statement_template).render(action_list=mapping["Action"], arn=resource_arn).rstrip()
+
+
+def _query_tagging_api(tag_filters: dict) -> Union[dict, None]:
+    """
+    Queries the Tagging API to retrieve the ARNs of every resource with the given tags.
+
+    NOTE: https://docs.aws.amazon.com/cli/latest/reference/resourcegroupstaggingapi/get-resources.html#get-resources
+
+    Parameters
+    ----------
+    tag_filters : dict
+        The tag filters to restrict output to only those resources with the given tags.
+
+        NOTE: https://docs.aws.amazon.com/cli/latest/reference/resourcegroupstaggingapi/get-resources.html#options
+
+    Returns
+    -------
+    dict
+        A ResourceTagMappingList, which contains a list of resource ARNs and tags associated with each.
+
+        NOTE: https://docs.aws.amazon.com/cli/latest/reference/resourcegroupstaggingapi/get-resources.html#output
+
+
+    """
+    try:
+        return boto3.client("resourcegroupstaggingapi").get_resources(TagFilters=tag_filters)["ResourceTagMappingList"]
+    except Exception as query_error:
+        LOG.error("Failed to call get_resources from resource group tagging api: %s", query_error)
+        return None
+
+
+def _generate_statement_list(tag_filters: dict) -> List[str]:
+    """
+    Generates a list of IAM statements in the form of YAML strings with default action permissions for resources with the given tags.
+
+    Parameters
+    ----------
+    tag_filters : dict
+        The tag filters to restrict output to only those resources with the given tags.
+
+        NOTE: https://docs.aws.amazon.com/cli/latest/reference/resourcegroupstaggingapi/get-resources.html#options
+
+    Returns
+    -------
+    List[str]
+        The list of IAM statements in the form of YAML strings corresponding to the resources specified by the given tags.
+    """
+
+    resource_tag_mapping_list = _query_tagging_api(tag_filters)
+
+    if resource_tag_mapping_list is None:
+        LOG.error("Failed to receive get_resources response, cannot generate any IAM statements.")
+        return ["# Failed to query tagging api, can't generate IAM permissions for your resources."]
```

Comment:
This seems like an unexpected return value from this function, we should rather return `None` or raise an exception so that caller can handle this error case.